resources/views/components/game-progress-bar.blade.php

```diff
@@ -0,0 +1,75 @@
+@props([
+    'awardIndicator' => null, // 'unfinished' | 'beaten-softcore' | 'beaten-hardcore' | 'completed' | 'mastered' | null
+    'softcoreProgress' => 0,
+    'hardcoreProgress' => 0,
+    'maxProgress' => 100,
+    'tooltip' => null, // ?string
+])
+
+<?php
+    $hardcoreProgressBarWidth = $softcoreProgressBarWidth = '0';
+
+    if ($maxProgress > 0) {
+        if ($softcoreProgress >= $maxProgress) {
+            $softcoreProgressBarWidth = sprintf("%01.2f", ($maxProgress - $hardcoreProgress) * 100 / $maxProgress);
+        } elseif ($softcoreProgress > 0) {
+            $softcoreProgressBarWidth = sprintf("%01.2f", ($softcoreProgress - $hardcoreProgress) * 100 / $maxProgress);
+        }
+
+        if ($hardcoreProgress >= $maxProgress) {
+            $hardcoreProgressBarWidth = '100.0';
+        } elseif ($hardcoreProgress > 0) {
+            $hardcoreProgressBarWidth = sprintf("%01.2f", $hardcoreProgress * 100 / $maxProgress);
+        }
+    }
+
+    if (!$tooltip) {
+        $tooltip = "{$softcoreProgress} ({$hardcoreProgress}) of {$maxProgress}";
```

Comment:
This looks awkward if only softcore achievements are unlocked
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/53000dbb-ec2b-46d3-bbf2-9c20386b4a0e)

Reply:
Agreed. I've changed the logic to:

```php
if (!$tooltip) {
    if ($hardcoreProgress === $softcoreProgress) {
        $tooltip = "{$hardcoreProgress} of {$maxProgress}";
    } else {
        $tooltip = "{$softcoreProgress} ({$hardcoreProgress} hardcore) of {$maxProgress}";
    }
}
```

How do you feel about this? Certainly open to additional adjustments from here.

Reply:
That won't help if `$softcore > 0` and `$hardcore == 0` (i.e. softcore only players), but it does make it better for hardcore only players.

Reply:
Though now the tooltip feels redundant with the text.

Reply:
Implementation changed to:

```php
if (!$tooltipLabel) {
    if ($hardcoreProgress === $softcoreProgress) {
        // Same progress for both modes.
        $tooltipLabel = "Progress: {$hardcoreProgress}/{$maxProgress} (hardcore)";
    } else if ($hardcoreProgress === 0 && $softcoreProgress > 0) {
        // Only softcore progress.
        $tooltipLabel = "Progress: {$softcoreProgress}/{$maxProgress} (softcore only)";
    } else {
        // Mixed progress.
        $tooltipLabel = "Progress: {$softcoreProgress}/{$maxProgress} (softcore), {$hardcoreProgress}/{$maxProgress} (hardcore)";
    }
}
```

In this case, while it may be duplicative if the user has single-mode progress, more context is given than the label if it's mixed mode. I think the tooltip should hang around.